s3/src/main/java/ch/cyberduck/core/sts/AssumeRoleWithWebIdentitySTSCredentialsConfigurator.java

```diff
@@ -48,13 +54,51 @@ public Credentials configure(final Host host) {
 
         AssumeRoleWithWebIdentityRequest webIdReq = new AssumeRoleWithWebIdentityRequest()
                 .withWebIdentityToken(credentials.getOauth().getAccessToken());
-//                .withPolicy("policy")
-//                .withRoleArn("consoleAdmin")
-//                .withRoleSessionName("testSession");
-//                .putCustomQueryParameter("Version", "2011-06-15");
 
-        if(log.isDebugEnabled()) {
-            log.debug("Assuming role with web identity for host: {}", host);
+        String durationsecondsProp = "s3.assumerole.durationseconds";
```

Comment:
Use `HostPreferences` and override properties in .cyberduckprofile.

Reply:
is our code correct now?
Now we also check if durationSeconds is a valid size (900s - 42300s). If it is out of bounds I throw a IllegalArgumentException. Is that fine?